lib/chef/knife/ec_restore.rb

```diff
@@ -238,11 +238,17 @@ def upload_org_data(name)
           # pivotal and members of the admins group. Use the same strategy
           # above here.
           #
-          ['admins', 'billing-admins', 'public_key_read_access'].each do |group|
+          groups = ['admins', 'billing-admins']
+          groups.push('public_key_read_access') if server.supports_public_key_read_access?
+
+          groups.each do |group|
             restore_group(chef_fs_config, group, :clients => false)
           end
 
-          ['/acls/groups/billing-admins.json', '/acls/groups/public_key_read_access.json'].each do |acl|
+          acls_groups_paths = ['/acls/groups/billing-admins.json']
+          acls_groups_paths.push('/acls/groups/public_key_read_access.json') if server.supports_public_key_read_access?
```

Comment:
Yes, looks like the rescue happens then it gets reported at the end.  Here's a backup of 12.1.0 with two orgs being restored to a 12.15.8:

```
knife ec restore backups/ --purge
Restoring users
WARNING: Skipping pivotal user.  To overwrite pivotal, pass --overwrite-pivotal.
Restoring organization[acmeinc]
Restoring org admin data
WARN: Could not find /groups/public_key_read_access.json on disk. Will not restore.
Copying /acls/groups/billing-admins.json
Copying /acls/groups/public_key_read_access.json
ERROR: /acls/groups/public_key_read_access.json ACLs cannot be deleted.
Restoring the rest of the org
Copying /clients
Copying /containers
Copying /cookbook_artifacts
Copying /cookbooks
Copying /data_bags
Copying /environments
Copying /invitations.json
Copying /members.json
Copying /nodes
Copying /org.json
Copying /policies
Copying /policy_groups
Copying /roles
Copying /groups/00000000000003a94e506949fa5629f5.json
Copying /groups/admins.json
Updated /groups/admins.json
Copying /groups/clients.json
Copying /groups/users.json
Copying /groups/00000000000003a94e506949fa5629f5.json
Copying /groups/admins.json
Updated /groups/admins.json
Copying /groups/clients.json
Copying /groups/users.json
Copying /acls/groups/00000000000003a94e506949fa5629f5.json
Copying /acls/groups/admins.json
Copying /acls/groups/clients.json
Copying /acls/groups/users.json
Copying /acls/clients
Copying /acls/containers
Copying /acls/environments
Copying /acls/organization.json
WARN: Could not find /groups/public_key_read_access.json on disk. Will not restore.
Restoring organization[brewinc]
Restoring org admin data
WARN: Could not find /groups/public_key_read_access.json on disk. Will not restore.
Copying /acls/groups/billing-admins.json
Copying /acls/groups/public_key_read_access.json
ERROR: /acls/groups/public_key_read_access.json ACLs cannot be deleted.
Restoring the rest of the org
Copying /clients
Copying /containers
Copying /cookbook_artifacts
Copying /cookbooks
Copying /data_bags
Copying /environments
Copying /invitations.json
Copying /members.json
Copying /nodes
Copying /org.json
Copying /policies
Copying /policy_groups
Copying /roles
Copying /groups/00000000000015096a9c2b30c5e076ba.json
Copying /groups/00000000000064af00f9de8836ae986c.json
Copying /groups/admins.json
Updated /groups/admins.json
Copying /groups/clients.json
Copying /groups/users.json
Copying /groups/00000000000015096a9c2b30c5e076ba.json
Copying /groups/00000000000064af00f9de8836ae986c.json
Copying /groups/admins.json
Updated /groups/admins.json
Copying /groups/clients.json
Copying /groups/users.json
Copying /acls/groups/00000000000015096a9c2b30c5e076ba.json
Copying /acls/groups/00000000000064af00f9de8836ae986c.json
Copying /acls/groups/admins.json
Copying /acls/groups/clients.json
Copying /acls/groups/users.json
Copying /acls/clients
Copying /acls/containers
Copying /acls/environments
Copying /acls/organization.json
WARN: Could not find /groups/public_key_read_access.json on disk. Will not restore.
Restoring user ACLs
WARNING: Skipping pivotal user.  To overwrite pivotal, pass --overwrite-pivotal.

Error Summary Report
{
  "timestamp": "2017-07-25 20:08:28 -0400",
  "message": "400 \"Bad Request\"",
  "backtrace": [
    "/opt/chefdk/embedded/lib/ruby/2.4.0/net/http/response.rb:122:in `error!'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/http.rb:152:in `request'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/http.rb:123:in `put'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:369:in `block in put_acl'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:368:in `each'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:368:in `put_acl'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:105:in `block in restore_user_acls'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:117:in `block in for_each_user'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:110:in `foreach'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:110:in `for_each_user'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:103:in `restore_user_acls'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:58:in `run'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/knife.rb:442:in `block in run_with_pretty_exceptions'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/local_mode.rb:44:in `with_server_connectivity'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/knife.rb:441:in `run_with_pretty_exceptions'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/knife.rb:219:in `run'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/application/knife.rb:156:in `run'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/bin/knife:25:in `<top (required)>'",
    "/opt/chefdk/bin/knife:264:in `load'",
    "/opt/chefdk/bin/knife:264:in `<main>'"
  ],
  "exception": "Net::HTTPServerException"
}{
  "timestamp": "2017-07-25 20:08:28 -0400",
  "message": "400 \"Bad Request\"",
  "backtrace": [
    "/opt/chefdk/embedded/lib/ruby/2.4.0/net/http/response.rb:122:in `error!'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/http.rb:152:in `request'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/http.rb:123:in `put'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:369:in `block in put_acl'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:368:in `each'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:368:in `put_acl'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:105:in `block in restore_user_acls'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:117:in `block in for_each_user'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:110:in `foreach'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:110:in `for_each_user'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:103:in `restore_user_acls'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:58:in `run'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/knife.rb:442:in `block in run_with_pretty_exceptions'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/local_mode.rb:44:in `with_server_connectivity'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/knife.rb:441:in `run_with_pretty_exceptions'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/knife.rb:219:in `run'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/application/knife.rb:156:in `run'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/bin/knife:25:in `<top (required)>'",
    "/opt/chefdk/bin/knife:264:in `load'",
    "/opt/chefdk/bin/knife:264:in `<main>'"
  ],
  "exception": "Net::HTTPServerException"
}{
  "timestamp": "2017-07-25 20:08:28 -0400",
  "message": "400 \"Bad Request\"",
  "backtrace": [
    "/opt/chefdk/embedded/lib/ruby/2.4.0/net/http/response.rb:122:in `error!'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/http.rb:152:in `request'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/http.rb:123:in `put'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:369:in `block in put_acl'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:368:in `each'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:368:in `put_acl'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:105:in `block in restore_user_acls'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:117:in `block in for_each_user'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:110:in `foreach'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:110:in `for_each_user'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:103:in `restore_user_acls'",
    "/Users/jmiller/.chefdk/gem/ruby/2.4.0/gems/knife-ec-backup-2.2.1/lib/chef/knife/ec_restore.rb:58:in `run'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/knife.rb:442:in `block in run_with_pretty_exceptions'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/local_mode.rb:44:in `with_server_connectivity'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/knife.rb:441:in `run_with_pretty_exceptions'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/knife.rb:219:in `run'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/application/knife.rb:156:in `run'",
    "/opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/bin/knife:25:in `<top (required)>'",
    "/opt/chefdk/bin/knife:264:in `load'",
    "/opt/chefdk/bin/knife:264:in `<main>'"
  ],
  "exception": "Net::HTTPServerException"
}

Error(s) Summary file located at: 'backups//errors/restore-2017-07-25T20:08:28-04:00.json'
```